### PR TITLE
Support context propagation with opentracing fallback.

### PR DESCRIPTION
**Pull request by Azuka**
I took a pass at allowing modification of the context before running tasks, as well as using it in all pre publish handlers on the server. This should not be a breaking change as opentracing will continue to be used as before, just initiated differently. The only potential panic I can think of is if someone calls `Set...Handler(nil)`.

Can someone keep me honest as far as the chord, chain and group implementations go? I have mostly just used a single task at a time. The server implementation is a bit ugly, so suggestions would be welcome as well.

This relates to discussions in #384.

### CODE REVIEW

File: v1/compat.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
+package machinery
+
+import (
+	"context"
+	"github.com/RichardKnop/machinery/v1/tasks"
```
Comment (marcocali) [inline v1/compat.go]: Please organize imports in 3 different groups:

- Library
- 3rd party
- Local
Same as described here: golang/tools@ed69e84

Comment (Azuka) [reply v1/compat.go]: Should be resolved, @marcocali

Comment (marcocali) [reply v1/compat.go]: No, it is not.

File: v1/compat.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+	taskSpan := tracing.StartSpanFromHeaders(signature.Headers, signature.Name)
+	tracing.AnnotateSpanWithSignatureInfo(taskSpan, signature)
+	return opentracing.ContextWithSpan(ctx, taskSpan)
+}
+
+// OpenTracingPostTaskHandler ends an opentracing span based on the passed in context
+func OpenTracingPostTaskHandler(ctx context.Context, _ *tasks.Signature) {
+	taskSpan := opentracing.SpanFromContext(ctx)
+	if taskSpan != nil {
+		taskSpan.Finish()
```
Comment (marcocali) [inline v1/compat.go]: Consider using `defer`.

File: v1/compat.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+	}
+}
+
+// OpenTracingPrePublishTaskHandler propagates opentracing information in outgoing task header information
+func OpenTracingPrePublishTaskHandler(ctx context.Context, signature *tasks.Signature) func() {
+	span, _ := opentracing.StartSpanFromContext(ctx, "SendTask", tracing.ProducerOption(), tracing.MachineryTag)
+
+	// tag the span with some info about the signature
+	signature.Headers = tracing.HeadersWithSpan(signature.Headers, span)
```
Comment (marcocali) [inline v1/compat.go]: No need for `signature != nil` check here?

**Comment by Azuka** [comment]
@RichardKnop do you mind taking a look at this? I've been using in production for more than a year now.

Also I'm not sure why codecov is reporting on files I haven't touched.

File: v1/compat.go (hunk 2 above)
Comment (Azuka) [reply v1/compat.go:51]: Not when it's the only call there.

File: v1/compat.go (hunk 3 above)
Comment (Azuka) [reply v1/compat.go:60]: It would likely have panic-ed earlier, I think.

**Comment by Azuka** [comment]
@RichardKnop, call you look?

**Pull request closed without merging**